File: search_data.R
require(RSiteCatalyst)

SCAuth("soumen.seth:FSN E-Commerce","770f388b78d019017d5e8bd7a63883fb")

#Website Data
i <- 1
raw_perm <- data.frame()
raw_temp <- data.frame(1)
nrow(raw_temp)
while (nrow(raw_temp) != 0){
  raw_temp <- data.frame()
  raw_temp <- QueueTrended("fsnecommerceprod",
                           '2016-02-01',
                           Sys.Date() - 1,
                           metrics =  'instances',
                           elements = "evar6",
                           start = i,
                           top = 50000,
                           date.granularity = "year",
                           interval.seconds = 5,
                           max.attempts = 100,
                           validate = TRUE)
  raw_perm <- rbind(raw_perm, raw_temp)
  i <- i + 50000
  timestamp();
  print(ceiling(i/50000)-1);
  print(nrow(raw_temp));
  print(nrow(raw_perm));
}
website_search_data <- raw_perm


#App Data
i <- 1
raw_perm <- data.frame()
raw_temp <- data.frame(1)
nrow(raw_temp)
while (nrow(raw_temp) != 0){
  raw_temp <- data.frame()
  raw_temp <- QueueTrended("fsnecommercemobileappprod",
                           '2016-02-01',
                           Sys.Date() - 1,
                           metrics =  'instances',
                           elements = "evar6",
                           start = i,
                           top = 50000,
                           date.granularity = "year",
                           interval.seconds = 5,
                           max.attempts = 100,
                           validate = TRUE)
  raw_perm <- rbind(raw_perm, raw_temp)
  i <- i + 50000
  timestamp();
  print(ceiling(i/50000)-1);
  print(nrow(raw_temp));
  print(nrow(raw_perm));
  
}
app_search_data <- raw_perm

File: search_data.py
import os
import os.path 
import arrow
import omniture
import sys
import IPython; embed = IPython.embed
PRINT_CONSTANTS = False
analytics = omniture.authenticate('soumen.seth:FSN E-Commerce', '770f388b78d019017d5e8bd7a63883fb')
dir_path = os.path.dirname(os.path.realpath(__file__))

suite = analytics.suites['fsnecommerceprod']

if PRINT_CONSTANTS:
  with open(os.path.join(dir_path, "analytics.suites.txt"), 'w') as f:
    f.write("%s" % suite)
  with open(os.path.join(dir_path, "metrics.txt"), 'w') as f:
    f.write("%s" % suite.metrics)
  with open(os.path.join(dir_path, "elements.txt"), 'w') as f:
    f.write("%s" % suite.elements)

  print("Wrote constants in files. Exiting now.")
  sys.exit()



startdate = arrow.now().replace(days=-30).format('YYYY-MM-DD')
enddate = arrow.now().replace(days=-1).format('YYYY-MM-DD')

report = suite.report \
		.element('evar6', top=50, startingWith=0) \
		.metric('instances') \
    .range(startdate, enddate)\
		.run()
		#.metric('orders') \
    #.granularity('day')\
print(" --- ")
print(report)

#print(report.keys())


